### ISSUE TITLE
[Due for payment 2025-06-11] Attendee tracking feature should be a sub-category of 'Rules'

### ISSUE THREAD

**Issue by Julesssss** [Expensify/App#50802]
_A part of collect v2_

### Summary of changes
- Make attendee tracking a feature that can be enabled/disabled via a toggle
- Make it restricted to Control policies, by placing the toggle in the Rules section of the workspace editor (which can only be accessed after upgrading)

<img width="1138" alt="Image" src="https://github.com/user-attachments/assets/4cc08b53-1a72-4399-8dda-a3b9fb21bddf" />
   
   - Text to use below the toggle: `Track the per-person cost for every expense.`

Initial discussion [here](https://expensify.slack.com/archives/C06ML6X0W9L/p1728987893006389?thread_ts=1728941642.251739&cid=C06ML6X0W9L) (internal).

### Technical changes
1. **Backend**
   - Create a new API command **`SetPolicyAttendeeTrackingEnabled`** that takes a `policyID: string` and `enabled: boolean`
   - Set the enabled state of the feature as a new policy attribute `isAttendeeTrackingEnabled`
   - https://github.com/Expensify/Expensify/issues/504051

2. **Frontend**
   - Update `onyx/Policy.ts` with a new prop `isAttendeeTrackingEnabled?: boolean`
   - Add a toggle to enable or disable Attendee Tracking in the Rules section of the workspace editor, under the existing `eReceipts` toggle
       - For backwards compatibility, assume the feature is disabled if the prop `isAttendeeTrackingEnabled` doesn't exist
       - Use the new API command `SetPolicyAttendeeTrackingEnabled` when toggling the option, with optimistic data
    - Update `shouldShowAttendees` to only show the attendee selector when the feature is enabled on the selected workspace

3. **Backend** (can be done as follow-up updates)
    1. Update commands to throw an error if trying to set/edit attendees on a collect policy expense
    2. Disable the flag after downgrading a policy
    3. Prevent from enabling the feature on collect policies

**Comment by dylanexpensify** [comment]
@Julesssss assigning you for this for now, but feel free to assign someone else if need be!

**Comment by Julesssss** [comment]
Project is held

**Comment by Julesssss** [comment]
Project remains held

**Comment by Julesssss** [comment]
FYI @francoisl. 

Perhaps you've already handled this elsewhere but there was a request to changed where the attendee feature switch lives.

**Comment by francoisl** [comment]
I don't think it's even a feature you can disable at the moment, right? I also don't see any mention of this in the design doc, but it sounds like it's something we can do as a follow-up once all the main functionality is ironed out.

**Comment by Julesssss** [comment]
Yeah this came later and wasn't added to the doc. I don't think the feature has a toggle in the settings, but if I recall correctly there was a high level toggle under categories with that disabled it. That could be outdated though 😬 I have no idea

**Comment by francoisl** [comment]
Updated the OP above, going to reprioritize to finish this before EOM.

**Comment by francoisl** [comment]
@allroundexperts is this something you're interested in working on as a follow-up improvement task?

**Comment by JmillsExpensify** [comment]
Done

**Comment by francoisl** [comment]
No, I'm finishing it up today, it should be available on staging early next week. I'll let you know when!

**Comment by allroundexperts** [comment]
Draft PR ready: https://github.com/Expensify/App/pull/62684

**Comment by francoisl** [comment]
One of the 2 backend PRs is deployed, the second one is still in review - ETA merged today and deployed tomorrow.

**Comment by danielrvidal** [comment]
Nice, this looks like it's live on staging! Here is a collect workspace. 

<img width="1436" alt="Image" src="https://github.com/user-attachments/assets/3b1bae8e-b6fa-48f4-b835-1623a8ebced3" />

**Comment by francoisl** [comment]
Coming from [this internal discussion](https://expensify.slack.com/archives/C049HHMV9SM/p1748958361354399), we're going to change the default behavior to make the feature enabled by default for existing policies that don't have the `isAttendeeTrackingEnabled` flag, to avoid confusion with how attendee tracking was always available on Classic. Will work on this [here](https://github.com/Expensify/App/issues/63464).

**Comment by joekaufmanexpensify** [comment]
Assigning @ahmedGaber93 as they were C+ reviewer here.

**Comment by joekaufmanexpensify** [comment]
Payment here is as follows:

- @ahmedGaber93 $250 for C+ review via NewDot.

**Comment by joekaufmanexpensify** [comment]
@ahmedGaber93 please request payment at your earliest convenience. Closing as this is otherwise all set!

**Comment by allroundexperts** [comment]
No payment due for me. I’ll be getting paid [here](https://github.com/Expensify/App/issues/64981)

**Comment by joekaufmanexpensify** [comment]
Sounds good. Updated summary to remove you. 👍

**Comment by trjExpensify** [comment]
approved $250 for @ahmedGaber93.
